quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -141,6 +145,91 @@ fn get_indexes_metadatas_handler(
         .map(make_response)
 }
 
+/// Describes an index with its main information and statistics.
+#[derive(Serialize, Deserialize, utoipa::ToSchema)]
+struct IndexStats {
+    pub index_id: String,
+    pub index_uri: Uri,
+    pub num_published_splits: usize,
+    pub num_published_docs: u64,
+    pub size_published_docs: Byte,
+    pub timestamp_field_name: Option<String>,
+    pub min_timestamp: Option<i64>,
+    pub max_timestamp: Option<i64>,
+}
+
+#[utoipa::path(
+    get,
+    tag = "Indexes",
+    path = "/indexes/{index_id}/describe",
+    responses(
+        (status = 200, description = "Successfully fetched stats about Index.", body = IndexStats)
+    ),
+    params(
+        ("index_id" = String, Path, description = "The index ID to describe."),
+    )
+)]
+
+/// Describes an index.
+async fn describe_index(
+    index_id: String,
+    metastore: Arc<dyn Metastore>,
+) -> Result<IndexStats, MetastoreError> {
+    let index_metadata = metastore.index_metadata(&index_id).await?;
+    let query = ListSplitsQuery::for_index(&index_id);
+    let splits = metastore.list_splits(query).await?;
+    let published_splits: Vec<Split> = splits
+        .into_iter()
+        .filter(|split| split.split_state == SplitState::Published)
+        .collect();
+    let mut total_num_docs = 0;
+    let mut total_bytes = 0;
+    let mut min_timestamp: Option<i64> = None;
+    let mut max_timestamp: Option<i64> = None;
+
+    for split in &published_splits {
+        total_num_docs += split.split_metadata.num_docs as u64;
+        total_bytes += split.split_metadata.footer_offsets.end;
+
+        for time_range in &split.split_metadata.time_range {
```

Comment:
This is not very readable.

What do you think of this?

```rust
if let Some(time_range) = &split.split_metadata.time_range {
    min_timestamp = min_timestamp.map(|ts| ts.min(*time_range.start()));
    max_timestamp = max_timestamp.map(|ts| ts.max(*time_range.end()));
}

```

Reply:
yeah the second for loop is unnecessary,  it is going to iterate on only 1 element. I tried your code it is giving me **null** output, So i am doing this instead  
`if let Some(time_range) = &split.split_metadata.time_range {
            min_timestamp = min_timestamp.min(Some(*time_range.start())).or(Some(*time_range.start()));
            max_timestamp = max_timestamp.max(Some(*time_range.end())).or(Some(*time_range.end()));
        }`

Reply:
yeah for loop is unnecessary, it is going to iterate on only 1 item. I tried your code, it is giving me **null** output, So i did this instead:

```rust
if let Some(time_range) = &split.split_metadata.time_range {
            min_timestamp = min_timestamp.min(Some(*time_range.start())).or(Some(*time_range.start()));
            max_timestamp = max_timestamp.max(Some(*time_range.end())).or(Some(*time_range.end()));
        }
```

Reply:
yeah for loop is unnecessary, it is going to iterate on only 1 item. I tried your code, it is giving me null output, So i did this instead:

```rust
if let Some(time_range) = &split.split_metadata.time_range {
    min_timestamp = min_timestamp.min(Some(*time_range.start())).or(Some(*time_range.start()));
    max_timestamp = max_timestamp.max(Some(*time_range.end())).or(Some(*time_range.end()));
}
```